### PR TITLE
git hub pull attach fails

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**
I get this:
```
~/src $ git hub pull attach 98 
Pushing branch to branch in origin
Attaching commits in branch branch to issue #98 (to be merged to owner/repo:master)
Error: GitHub error: Invalid request.

For 'properties/issue', "98" is not an integer.

```

### CODE REVIEW

**Comment by leandro-lucarella-sociomantic** [comment]
This is probably due to https://developer.github.com/changes/2018-09-25-stricter-validation-coming-soon-in-the-rest-api/ and some missing conversions.

Can you attach the output of `git hub -v pull attach 98`?

**Comment by leandro-lucarella-sociomantic** [comment]
Fix attached. I attached the code using `git-hub` with the fix applied :laughing: , so I guess the fix should work, but if you can, please do test it too.

**Comment by leandro-lucarella-sociomantic** [comment]
Fix attached. I attached the code using `git-hub` with the fix applied :laughing: , so I guess the fix should work, but if you can, please do test it too.

**Comment by leandro-lucarella-sociomantic** [comment]
@daniel-zullo-sociomantic can you test this?

File: git-hub [hunk 1]
```diff
@@ -1142,7 +1142,7 @@ class IssueCmd (CmdGroup):
         @classmethod
         def setup_parser(cls, parser):
             parser.add_argument('issues',
-                nargs='+', metavar=cls.id_var,
+                nargs='+', metavar=cls.id_var, type=int,
```
Comment (daniel-zullo) [inline git-hub:1145]: This change does not work and it isn't actually needed

```
./git-hub issue show 902
Traceback (most recent call last):
  File "../git-hub/git-hub", line 2156, in <module>
    main()
  File "../git-hub/git-hub", line 2150, in main
    args.run(parser, args)
  File "../git-hub/git-hub", line 649, in check_config_and_run
    cmd.run(parser, args)
  File "../git-hub/git-hub", line 1155, in run
    issue = req.get(cls.url(n))
  File "../git-hub/git-hub", line 1075, in url
    s += '/' + number
TypeError: cannot concatenate 'str' and 'int' objects
```

File: git-hub [hunk 2]
```diff
@@ -1179,7 +1179,7 @@ class IssueCmd (CmdGroup):
                 metavar='USER',
                 help="assign a user to the %s; must be a "
                 "valid GitHub login name" % cls.name)
-            parser.add_argument('-M', '--milestone', metavar='ID',
+            parser.add_argument('-M', '--milestone', metavar='ID', type=int,
```
Comment (daniel-zullo) [inline git-hub:1182]: This one works fine

File: git-hub [hunk 3]
```diff
@@ -1195,7 +1195,7 @@ class IssueCmd (CmdGroup):
         cmd_help = "update an existing issue"
         @classmethod
         def setup_parser(cls, parser):
-            parser.add_argument('issue', metavar=cls.id_var,
+            parser.add_argument('issue', metavar=cls.id_var, type=int,
```
Comment (daniel-zullo) [inline git-hub:1198]: Works

File: git-hub [hunk 4]
```diff
@@ -1270,7 +1270,7 @@ class IssueCmd (CmdGroup):
         cmd_help = "add a comment to an existing issue"
         @classmethod
         def setup_parser(cls, parser):
-            parser.add_argument('issue', metavar=cls.id_var,
+            parser.add_argument('issue', metavar=cls.id_var, type=int,
```
Comment (daniel-zullo) [inline git-hub:1273]: works

File: git-hub [hunk 5]
```diff
@@ -1287,7 +1287,7 @@ class IssueCmd (CmdGroup):
         cmd_help = "close an opened issue"
         @classmethod
         def setup_parser(cls, parser):
-            parser.add_argument('issue', metavar=cls.id_var,
+            parser.add_argument('issue', metavar=cls.id_var, type=int,
```
Comment (daniel-zullo) [inline git-hub:1290]: it does not work
```
 ../git-hub/git-hub issue close 908
Traceback (most recent call last):
  File "../git-hub/git-hub", line 2156, in <module>
    main()
  File "../git-hub/git-hub", line 2150, in main
    args.run(parser, args)
  File "../git-hub/git-hub", line 649, in check_config_and_run ../git-hub/git-hub issue close 908
Traceback (most recent call last):
  File "../git-hub/git-hub", line 2156, in <module>
    main()
  File "../git-hub/git-hub", line 2150, in main
    args.run(parser, args)
  File "../git-hub/git-hub", line 649, in check_config_and_run
    cmd.run(parser, args)
  File "../git-hub/git-hub", line 1308, in run
    issue = req.patch(cls.url(args.issue), state='closed')
  File "../git-hub/git-hub", line 1075, in url
    s += '/' + number
TypeError: cannot concatenate 'str' and 'int' objects

    cmd.run(parser, args)
  File "../git-hub/git-hub", line 1308, in run
    issue = req.patch(cls.url(args.issue), state='closed')
  File "../git-hub/git-hub", line 1075, in url
    s += '/' + number
TypeError: cannot concatenate 'str' and 'int' objects
```

File: git-hub [hunk 6]
```diff
@@ -1538,7 +1538,7 @@ class PullCmd (IssueCmd):
         cmd_help = "checkout the remote branch (head) of the pull request"
         @classmethod
         def setup_parser(cls, parser):
-            parser.add_argument('pull',
+            parser.add_argument('pull', type=int,
```
Comment (daniel-zullo) [inline git-hub:1541]: It does not work 
```
../git-hub/git-hub pull checkout 907
Traceback (most recent call last):
  File "../git-hub/git-hub", line 2156, in <module>
    main()
  File "../git-hub/git-hub", line 2150, in main
    args.run(parser, args)
  File "../git-hub/git-hub", line 649, in check_config_and_run
    cmd.run(parser, args)
  File "../git-hub/git-hub", line 1548, in run
    pull = req.get(cls.url(args.pull))
  File "../git-hub/git-hub", line 1075, in url
    s += '/' + number
TypeError: cannot concatenate 'str' and 'int' objects
```

File: git-hub [hunk 7]
```diff
@@ -1487,7 +1487,7 @@ class PullCmd (IssueCmd):
                 "to a pull request)"
         @classmethod
         def setup_parser(cls, parser):
-            parser.add_argument('issue', metavar='ISSUE',
+            parser.add_argument('issue', metavar='ISSUE', type=int,
```
Comment (daniel-zullo) [inline git-hub:1490]: works

**Review by daniel-zullo — CHANGES_REQUESTED** [review]
Many of the changes do not work.

You might need to fix update the `url()` method for the changes to work by converting the number to a string
https://github.com/sociomantic-tsunami/git-hub/blob/v1.0.x/git-hub#L1071-L1076

**Comment by daniel-zullo** [comment]
```diff
diff --git a/git-hub b/git-hub
index b0f186d..5d83ffe 100755
--- a/git-hub
+++ b/git-hub
@@ -1072,7 +1072,7 @@ class IssueUtil (object):
     def url(cls, number=None):
         s = '/repos/%s/%s' % (config.upstream, cls.gh_path)
         if number:
-            s += '/' + number
+            s += '/' + str(number)
         return s
 
     @classmethod
```

**Comment by leandro-lucarella-sociomantic** [comment]
Thanks for the testing! Yeah, I will have another look and do the conversion where it makes sense, I think it is better to use an `int` to store the PR/issue IDs as it also means we have some basic check on the args parser.

**Comment by leandro-lucarella-sociomantic** [comment]
Updated.

**Review by daniel-zullo — APPROVED** [review]
LGTM

**Comment by leandro-lucarella-sociomantic** [comment]
v1.0.3 released

**Pull request merged**